compiler/quilt/tools/core.py

```diff
@@ -73,6 +74,7 @@ def __init__(self, hashes, format, metadata=None):
         assert isinstance(hashes, list)
         assert isinstance(format, string_types), '%r' % format
         assert isinstance(metadata, dict)
+        assert not os.path.isabs(metadata.get('q_path') or '')
```

Comment:
`metadata.get('q_path', '')`

Reply:
Intentional -- this also handles the case where someone sets `'q_path'` to None, but `metadata.get('q_path', '')` doesn't.